app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -59,16 +62,35 @@ class ReviewRegistrationController @Inject() (
       if (request.registration.isCheckAndSubmitReady)
         request.registration.incorpJourneyId match {
           case Some(journeyId) =>
-            for {
-              incorporationDetails <- incorpIdConnector.getDetails(journeyId)
-              _                    <- markRegistrationAsReviewed()
-            } yield Ok(page(request.registration, incorporationDetails))
+            request.registration.organisationDetails.organisationType match {
+              case Some(UK_COMPANY)  => ukCompanyReview(journeyId)
+              case Some(SOLE_TRADER) => soleTraderReview(journeyId)
+              case _ =>
+                throw new InternalServerException(s"Company type not supported")
+            }
+
           case _ => Future(Redirect(routes.RegistrationController.displayPage()))
         }
       else
         Future(Redirect(routes.RegistrationController.displayPage()))
     }
 
+  private def soleTraderReview(journeyId: String)(implicit request: JourneyRequest[AnyContent]) =
+    for {
+      soleTraderDetails <- soleTraderInorpIdConnector.getDetails(journeyId)
+      _                 <- markRegistrationAsReviewed()
+    } yield Ok(
+      page(registration = request.registration, soleTraderDetails = Some(soleTraderDetails))
+    )
+
+  private def ukCompanyReview(journeyId: String)(implicit request: JourneyRequest[AnyContent]) =
+    for {
+      incorporationDetails <- incorpIdConnector.getDetails(journeyId)
+      _                    <- markRegistrationAsReviewed()
+    } yield Ok(
+      page(registration = request.registration, incorporationDetails = Some(incorporationDetails))
+    )
+
   private def markRegistrationAsReviewed()(implicit
```

Comment:
We may have to review this as well, as we are potentially going to have a lot of repetitive/duplicated code.
WE should potentially have a entityDetails (with potentially some optionals there) and in that case the code get a lot more elegant :thinking: 

We would then push difference to views :thinking:

Reply:
ok, just did enough to make sole trader work, not sure how other entities will look like yet, once know more we can bring common parts together